src/internal/connector/onedrive/api/drive.go

```diff
@@ -269,3 +271,76 @@ func (p *siteDrivePager) GetFolderIDByName(ctx context.Context, driveID, folderN
 
 	return empty, nil
 }
+
+// ---------------------------------------------------------------------------
+// Drive Paging
+// ---------------------------------------------------------------------------
+
+// DrivePager pages through different types of drive owners
+type DrivePager interface {
+	GetPage(context.Context) (api.PageLinker, error)
+	SetNext(nextLink string)
+	ValuesIn(api.PageLinker) ([]models.Driveable, error)
+}
+
+// GetAllDrives fetches all drives for the given pager
+func GetAllDrives(
+	ctx context.Context,
+	pager DrivePager,
+	retry bool,
+	maxRetryCount int,
+) ([]models.Driveable, error) {
+	drives := []models.Driveable{}
+
+	if !retry {
+		maxRetryCount = 0
+	}
+
+	// Loop through all pages returned by Graph API.
+	for {
+		var (
+			err  error
+			page api.PageLinker
+		)
+
+		// Retry Loop for Drive retrieval. Request can timeout
+		for i := 0; i <= maxRetryCount; i++ {
```

Comment:
do we still need this retry logic or is it covered by middleware retry handler we now have?

Reply:
I guess that depends on the pager implementation.  It *should* be an expectation we can carry through, but not a guarantee.  I'm fine removing this, but would prefer to do so in a follow-up PR.